src/common/wincmn.cpp

```diff
@@ -2907,10 +2904,7 @@ wxWindowBase::FromDIP(const wxSize& sz, const wxWindowBase* w)
 
     const int baseline = wxDisplay::GetStdPPIValue();
 
-    // Take care to not scale -1 because it has a special meaning of
-    // "unspecified" which should be preserved.
-    return wxSize(sz.x == -1 ? -1 : wxMulDivInt32(sz.x, dpi.x, baseline),
-                  sz.y == -1 ? -1 : wxMulDivInt32(sz.y, dpi.y, baseline));
+    return wxRescaleCoord(sz).From(baseline, baseline).To(dpi);
```

Comment:
You could use `GetStdPPI` instead of `GetStdPPIValue()`.

Reply:
Thanks, changed.